vehicle/remote_control_process.py

```diff
@@ -210,12 +217,65 @@ def __init__(self, remote_to_main_lock, main_to_remote_lock, remote_to_main_stri
                             datefmt=_LOGGER_DATE_FORMAT,
                             level=_LOGGER_LEVEL,
                             logger=self.logger)
+        self.gps = gps.GPS(self.logger, self.simulated_hardware)
+        self.default_navigation_parameters = NavigationParameters(travel_speed=0.0,
+                                                                  path_following_direction=Direction.EITHER,
+                                                                  vehicle_travel_direction=Direction.EITHER,
+                                                                  loop_path=True)
+        self.default_path_control_vals = PathControlValues(angular_p=0.9,
+                                                           lateral_p=-0.25,
+                                                           angular_d=0.3,
+                                                           lateral_d=-0.05)
+        self.nav_path = PathSection(points=[],
+                                    control_values=self.default_path_control_vals,
+                                    navigation_parameters=self.default_navigation_parameters,
+                                    max_dist=_MAXIMUM_ALLOWED_DISTANCE_METERS,
+                                    max_angle=_MAXIMUM_ALLOWED_ANGLE_ERROR_DEGREES,
+                                    end_dist=1.0,
+                                    end_angle=30)
+        self.maximum_velocity = _DEFAULT_MAXIMUM_VELOCITY
+        self.nav_path_list = []
+        self.nav_path_index = 0
+        self.gps_path = []
+        self.load_path_time = time.time()
+        self.loaded_path_name = ""
+        self.autonomy_hold = True
+        self.control_state = CONTROL_STARTUP
+        self.motor_state = STATE_DISCONNECTED
+        self.driving_direction = 1.0
+        self.gps_path_lateral_error = 0
+        self.gps_path_angular_error = 0
+        self.gps_error_update_time = 0
+        self.gps_angle_error_rate_averaging_list = []
+        self.gps_lateral_error_rate_averaging_list = []
+        self.last_autonomy_steer_cmd = 0
+        self.last_autonomy_strafe_cmd = 0
+        self.solution_age_averaging_list = []
+        self.voltage_average = 0
+        self.disengagement_time = time.time()
+
+        self.power_consumption_list = []
+        self.avg_watts_per_meter = 0
+        self.watt_hours_per_meter = 0
+        self.total_watts = 0
+        self.voltages = []
+        self.bus_currents = []
+        self.last_energy_segment = None
+        self.temperatures = []
+        self.last_calculated_steering = calculate_steering(0, 0, 0, _STEERING_ANGLE_LIMIT_DEGREES)
+        self.last_wifi_restart_time = time.time()
+        self.last_vel_cmd = 0
+        self.vel_cmd = 0
+
+        self.loop_count = -1
```

Comment:
This was a variable in `run_loop`. Changed to a field to avoid passing it all around to the extracted methods.